controllers/release/adapter.go

```diff
@@ -374,6 +384,19 @@ func (a *adapter) finalizeRelease() error {
 	}
 
 	if pipelineRun != nil {
+		// The finalizer could still exist at this point. It should be removed if the Release is finalized
```

Comment:
I'm a little confused when we would expect this to happen. Shouldn't the EnsureReleaseProcessingIsTracked operation always remove it?

Reply:
Not if we didn't get the status update or if the operation is disabled.

Reply:
Well, the operation isn't disabled. I am not sure why we would ever miss the status update. Nonetheless, can you at least put that in the comment (what your reply just was)?